src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String PATTERN = "dd.MM.yyyy";
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern(PATTERN);
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dayFrom = LocalDate.parse(dateFrom, FORMATTER);
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ").append(dateFrom).append(" - ").append(dateTo)
+                .append(System.lineSeparator());
+        int count = 0;
+        for (String name : names) {
+            int salary = 0;
+            count++;
+            for (String item : data) {
+                String[] processData = item.split(" ");
+                if (processData[1].equals(name)) {
+                    if (!(result.toString().contains(name))) {
+                        result.append(name).append(" - ");
+                    }
```

Comment:
you can rewrite your logic, so you could check the name with the date in the same condition